Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -33,6 +33,21 @@ public byte[] ToArray()
             return ((MemoryStream)writer.BaseStream).ToArray();
         }
 
+        // Gets the serialized data in an ArraySegment<byte>
+        // this is similar to ToArray(),  but it gets the data in O(1)
+        // and without allocations.
+        // Do not write anything else or modify the NetworkWriter
+        // while you are using the ArraySegment
+        public ArraySegment<byte> ToArraySegment()
+        {
+            writer.Flush();
+            if (((MemoryStream)writer.BaseStream).TryGetBuffer(out ArraySegment<byte> data))
+            { 
+                return data;
+            }
+            return new ArraySegment<byte>(ToArray());
```

Comment:
probably better to throw an error here, since it means we didn't create the memorystream the right way. and users would assume toarraysegment does no alloc, but it does

Reply:
ok,  testing